api/server/middleware_auth.go

```diff
@@ -38,6 +38,76 @@ func NewAuthMiddleware() *authMiddleware {
 type authMiddleware struct {
 }
 
+// newSecurityMiddleware based on auth configuration returns securityHandler or just
+func newSecurityMiddleware(authenticators map[string]auth.Authenticator) func(next http.HandlerFunc) http.HandlerFunc {
+	if auth.Enabled() {
+		return func(next http.HandlerFunc) http.HandlerFunc {
+			return securityHandler(authenticators, next)
+		}
+	}
+
+	return func(next http.HandlerFunc) http.HandlerFunc {
+		return next
+	}
+}
+
+// securityHandler implements Authentication and Authorization check at the same time
+// this functionality where not moved to separate functions because of simplicity
+func securityHandler(authenticators map[string]auth.Authenticator, next http.HandlerFunc) http.HandlerFunc {
+	if authenticators == nil {
+		return next
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		tokenHeader := r.Header.Get("Authorization")
+		tokens := strings.Split(tokenHeader, " ")
+
+		if len(tokens) < 2 {
+			w.WriteHeader(http.StatusForbidden)
+			return
+		}
+		token := tokens[1]
+
+		// Determine issuer
+		issuer, err := auth.TokenIssuer(token)
+		if err != nil {
+			w.WriteHeader(http.StatusForbidden)
+			return
+		}
+
+		// Authenticate user
+		ctx := r.Context()
```

Comment:
r.Context() is not used. ctx is supposed to have saved the token information in gRPC format.